domains/transfer/Transfer.API/API.IntegrationTests/Claiming/Api/Controllers/ClaimControllerTest.cs

```diff
@@ -0,0 +1,154 @@
+
+        var result = await client.DeleteAsync("api/claims/stop-claim-process");
+        result.StatusCode.Should().Be(HttpStatusCode.NotFound);
+    }
+
+    [Fact]
+    public async Task StartProcess_WhenNoClaimProcessHasStarted_ReturnsCreatedAt()
+    {
+        var subject = Guid.NewGuid();
+        var client = factory.CreateAuthenticatedClient(sub: subject.ToString());
+
+        var result = await client.PostAsync("api/claims/start-claim-process", null);
+        result.StatusCode.Should().Be(HttpStatusCode.Created);
+
+    }
+
+    [Fact]
+    public async Task StartProcess_WhenClaimProcessAlreadyStarted_ReturnsOk()
+    {
+        var subject = Guid.NewGuid();
+
+        await factory.SeedClaims(new List<ClaimSubject>()
+        {
+            new(subject)
+        });
+
+        var client = factory.CreateAuthenticatedClient(sub: subject.ToString());
+
+        var result = await client.PostAsync("api/claims/start-claim-process", null);
+        result.StatusCode.Should().Be(HttpStatusCode.OK);
+    }
+
+    [Fact]
+    public async Task GetClaimSubjectHistory_WhenHistoryEntryHasDifferentSubjectId_ReturnsNotFound()
+    {
+        var historyEntry = new ClaimSubjectHistory()
+        {
+            ActorId = Guid.NewGuid().ToString(),
+            ActorName = "Peter Producent",
+            AuditAction = "Insert",
+            CreatedAt = DateTimeOffset.UtcNow,
+            Id = Guid.NewGuid(),
+            SubjectId = Guid.NewGuid()
+        };
+        await factory.SeedClaimSubjectHistory(new List<ClaimSubjectHistory>()
+        {
+            historyEntry
+        });
+
+        var client = factory.CreateAuthenticatedClient(sub: Guid.NewGuid().ToString());
+        var result = await client.GetAsync("api/claims/claim-process-history");
+        result.StatusCode.Should().Be(HttpStatusCode.NotFound);
+    }
+
+    [Fact]
+    public async Task GetClaimSubjectHistory_WhenHistoryExists_ReturnsOK()
+    {
+        var subject = Guid.NewGuid();
+        var historyEntry = new ClaimSubjectHistory()
+        {
+            ActorId = Guid.NewGuid().ToString(),
+            ActorName = "Peter Producent",
+            AuditAction = "Insert",
+            CreatedAt = DateTimeOffset.UtcNow,
+            Id = Guid.NewGuid(),
+            SubjectId = subject
+        };
+        await factory.SeedClaimSubjectHistory(new List<ClaimSubjectHistory>()
+        {
+            historyEntry
+        });
+
+        var client = factory.CreateAuthenticatedClient(sub: subject.ToString());
+        var result = await client.GetFromJsonAsync<ClaimSubjectHistoryEntriesDto>("api/claims/claim-process-history");
+
+        result.Should().NotBeNull();
+        result!.History.Should().HaveCount(1);
+        result.History[0].ActorName.Should().Be(historyEntry.ActorName);
+        result.History[0].CreatedAt.Should().BeSameDateAs(historyEntry.CreatedAt);
+        result.History[0].AuditAction.Should().Be(historyEntry.AuditAction);
```

Comment:
Have you considered testing this by not directly inserting data into the history-table? This does not really test if the history entries are generated when doing changes on the API.

Reply:
When doing so i am running into some flaky test where history isnt always  created before making the request - do you have any ideas on how to solve it?

Reply:
fixed eb582eb2

Reply:
and e48a3de2